src/Database/Query.php

```diff
@@ -1412,11 +1413,17 @@ public function values($data)
      *
      * Can be combined with set() and where() methods to create update queries.
      *
-     * @param string $table The table you want to update.
+     * @param string|\Cake\Database\ExpressionInterface $table The table you want to update.
      * @return $this
      */
     public function update($table)
     {
+        if (!is_string($table) && !($table instanceof ExpressionInterface)) {
+            $text = 'Table must be of type string or "%s", got "%s"';
+            $message = sprintf($text, ExpressionInterface::class, gettype($table));
+            throw new InvalidArgumentException($message);
```

Comment:
example: `$query->update(['comments'])` would pass:
```
array(1) {
  [0]=>
  array(1) {
    [0]=>
    string(8) "comments"
  }
}

```
 to $table on the _buildUpdatePart so essentially arrays cannot currently be used